### PR TITLE
Log trace errors to Windows Event Log

### PR DESCRIPTION
**Pull request by sdmaclea**
Draft change to remove GUI dialog and add tracing to Windows Event Log for GUI AppHost

Starting to test now.... 

I considered enabling the Windows Event Log for all hosts regardless of GUI bit.  At this time it seemed more conservative to leave only for AppHost w/ GUI.

### CODE REVIEW

**Comment by sdmaclea** [comment]
```
The description for Event ID 1024 from source StandaloneApp.exe cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.

If the event originated on another computer, the display information had to be saved with the event.

The following information was included with the event: 

This executable is not bound to a managed DLL to execute. The binding value is: 'c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2'
A fatal error was encountered. This executable was not bound to load a managed DLL.
```

**Comment by sdmaclea** [comment]
I need to add the message list....

File: src/corehost/corehost.cpp [hunk 1]
```diff
@@ -293,25 +293,9 @@ int main(const int argc, const pal::char_t* argv[])
...
-            // unless the caller explicitly redirects stderr. This leads to bad experience of running the GUI app and nothing happening.
-            trace::set_error_writer(buffering_trace_writer);
-        }
-        else
-        {
-            trace::verbose(_X("Gui errors disabled, keeping errors in stderr."));
-        }
+        trace::verbose(_X("Redirecting errors to custom writer."));
+        // If this is a GUI application, buffer errors to use them later.
+        trace::set_error_writer(buffering_trace_writer);
```
Comment (vitek-karas) [inline src/corehost/corehost.cpp]: In this case we should probably write the errors to stderr as well - for example WPF designer would really like that approach. It's also helpful for any kind of test infra.

File: src/corehost/corehost.cpp [hunk 2]
```diff
@@ -293,25 +293,9 @@ int main(const int argc, const pal::char_t* argv[])
 #if defined(_WIN32) && defined(FEATURE_APPHOST)
     if (get_windows_graphical_user_interface_bit())
```
Comment (jkotas) [inline src/corehost/corehost.cpp]: I think we should write it to eventlog always; not just for the GUI apps.

File: src/corehost/corehost.cpp (hunk 1 above)
Comment (sdmaclea) [reply src/corehost/corehost.cpp]: Done

**Comment by sdmaclea** [comment]
This is what it currently will look like ...
image

**Comment by sdmaclea** [comment]
arm/arm64 are failing with unresolved symbols
```
4>corehost.obj : error LNK2019: unresolved external symbol __imp_DeregisterEventSource referenced in function wmain [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
4>corehost.obj : error LNK2019: unresolved external symbol __imp_RegisterEventSourceW referenced in function wmain [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
4>corehost.obj : error LNK2019: unresolved external symbol __imp_ReportEventW referenced in function wmain [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
4>F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\Release\apphost.exe : fatal error LNK1120: 3 unresolved externals [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
corehost.obj : error LNK2019: unresolved external symbol __imp_DeregisterEventSource referenced in function wmain [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
corehost.obj : error LNK2019: unresolved external symbol __imp_RegisterEventSourceW referenced in function wmain [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
corehost.obj : error LNK2019: unresolved external symbol __imp_ReportEventW referenced in function wmain [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\Release\apphost.exe : fatal error LNK1120: 3 unresolved externals [F:\workspace\_work\1\s\bin\obj\win-arm64.Release\corehost\cli\apphost\apphost.vcxproj] [F:\workspace\_work\1\s\src\corehost\build.proj]
F:\workspace\_work\1\s\src\corehost\build.proj(74,5): error MSB3073: The command "F:\workspace\_work\1\s\src\corehost\build.cmd Release arm64 apphostver 3.0.0-preview7-27825-18 hostver 3.0.0-preview7-27825-18 fxrver 3.0.0-preview7-27825-18 policyver 3.0.0-preview7-27825-18 commit d04ab337fb387e8d705620b8974113e2fe53eff5 rid win-arm64 portable" exited with code 1.
```

Probably need an extra library reference for these platforms. Not sure which one yet...

**Pull request merged**